### ISSUE TITLE
Issue with sharepoint access - bad request error

### ISSUE THREAD

**Issue by abhpuur** [OneDrive/onedrive-sdk-ios#147]
I am able to access one drive for business using the sample poc iOSExplorer. But when I use the following code to access a sharepoint site, after the successful authentication (auth is using okta login 2 step verify), I do not see permissions screen but get an error "Sorry, but we're having trouble signing you in. We received a bad request".

Here are the details:
in appdelegate ODClient is set using:
[ODClient setActiveDirectoryAppId:@"a1467700-854d-4426-8caf-18a9e6a735f2"
                                resourceId:@"https://_<companyname>_.sharepoint.com/"
                                apiEndpoint:@"https://_<companyname>_.sharepoint.com/_api/v2.0/"
                                redirectURL:@"iOSExplorer_ES://com.abhi.iOSExplorerES"];

All the permissions are provided in azure portal and the appid and redirect URL match with the values above. What could be the issue? The documentation on using sharepoint with onedrivesdk can definitely be improved!

**Comment by kevklam** [comment]
The resourceId and apiEndpoint you are specifying are not for a valid tenant.

First make sure you've followed the steps under "Register your app for OneDrive for Business" at the following link:
https://dev.onedrive.com/app-registration.htm

If you want the app to work only for a specific tenant, then specify that tenant for the resourceId/apiEndpoint. Otherwise, do not specify them:

``` objc
[ODClient setActiveDirectoryAppId:@"a1467700-854d-4426-8caf-18a9e6a735f2"
redirectURL:@"iOSExplorer_ES://com.abhi.iOSExplorerES"];
```

Hope this helps.

-Kevin

**Comment by abhpuur** [comment]
Hi Kevin: As I mentioned I am trying to show contents of a sharepoint site and not onedrive. Sorry the endpoint and resourceid got changed in comment while I was trying to italicize here. I can't specifiy the company name here but its a valid tenant and I had followed the steps provided for app registration.

resourceId:@"https://<companyname>.sharepoint.com/"
apiEndpoint:@"https://<companyname>.sharepoint.com/_api/v2.0/"

Also, when I try to connect to a different sharepoint to which the user logging in has group admin privilege it works ok. could it be a permissions issue? Another important difference is in the first case the login is via Okta and after that is successful, the permissions screen never appears.
